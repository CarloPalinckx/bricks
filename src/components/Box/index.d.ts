import { SFC } from 'react';
import { TrblType } from '../../utility/trbl';
declare type PropsType = JSX.IntrinsicElements['div'] & {
    justifyContent?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'space-between' | 'space-around' | 'space-evenly';
    alignItems?: 'flex-start' | 'flex-end' | 'center' | 'stretch';
    alignContent?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'space-between' | 'space-around';
    direction?: 'row' | 'row-reverse' | 'column' | 'column-reverse';
    inline?: boolean;
    height?: string;
    width?: string;
    margin?: TrblType;
    padding?: TrblType;
    maxHeight?: string;
    minHeight?: string;
    maxWidth?: string;
    minWidth?: string;
    wrap?: boolean;
    grow?: number;
    shrink?: number;
    basis?: string;
    order?: number;
    alignSelf?: 'auto' | 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch';
    position?: 'static' | 'relative' | 'fixed' | 'absolute';
    top?: string;
    right?: string;
    bottom?: string;
    left?: string;
};
declare const Box: SFC<PropsType>;
export { PropsType };
export default Box;
