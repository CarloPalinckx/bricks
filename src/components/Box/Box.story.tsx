import { boolean, number, select, text } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';
import React from 'react';
import styled from 'styled-components';
import trbl from '../../utility/trbl';
import Spacer from '../Spacer/Spacer.style';
import Text from '../Text';
import Box from './Box.style';

const Wrapper = styled.div`
    border: 1px dashed #aaa;
    height: 90vh;
`;

const Item = styled.div`
    background: #eee;
    border: 1px dashed #aaa;
    width: 100%;
    height: 100%;
`;

const justifyOptions = ['flex-start', 'flex-end', 'center', 'stretch', 'space-between', 'space-around', 'space-evenly'];
const alignItemsOptions = ['flex-start', 'flex-end', 'center', 'stretch'];
const alignContentOptions = ['flex-start', 'flex-end', 'center', 'stretch', 'space-between', 'space-around'];
const alignSelf = ['auto', 'flex-start', 'flex-end', 'center', 'baseline', 'stretch'];
const directionOptions = ['row', 'row-reverse', 'column', 'column-reverse'];

storiesOf('Box', module)
    .add('Default', () => {
        const items:Array<JSX.Element> = [];

        for (let i = 0; i < 6; i++) {
            items.push(
                <Box
                    key={i}
                    margin={trbl(select('margin', [0, 6, 12], 0, 'Child'))}
                    grow={number('grow', '1', 'Child')}
                    shrink={number('shrink', '1', 'Child')}
                    basis={text('basis', 'auto', 'Child')}
                    order={number('order', '1', 'Child')}
                    alignSelf={select('alignSelf', alignSelf, alignSelf[0], 'Child')}
                >
                    <Item>
                        <Spacer offsetType="inner" offset={trbl(48)}>
                            <Text>Box {i + 1}</Text>
                        </Spacer>
                    </Item>
                </Box>
            );
        }

        return (
            <Wrapper>
                <Box
                    margin={trbl(select('margin', [-12, -6, 0, 6, 12], 0, 'Parent'))}
                    justifyContent={select('justifyContent', justifyOptions, justifyOptions[0], 'Parent')}
                    alignItems={select('alignItems', alignItemsOptions, alignItemsOptions[0], 'Parent')}
                    alignContent={select('alignContent', alignContentOptions, alignContentOptions[0], 'Parent')}
                    direction={select('direction', directionOptions, directionOptions[0], 'Parent')}
                    wrap={boolean('wrap', true, 'Parent')}
                    height="100%"
                >
                    {items}
                </Box>
            </Wrapper>
        );
    });
