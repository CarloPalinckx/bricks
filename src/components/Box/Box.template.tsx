import React, { StatelessComponent } from 'react';
import { StyledType } from '../../utility/styled';
import { TrblType } from '../../utility/trbl';

type PropsType = StyledType & {
    justifyContent?:'flex-start'|'flex-end'|'center'|'stretch'|'space-between'|'space-around'|'space-evenly';
    alignItems?:'flex-start'|'flex-end'|'center'|'stretch';
    alignContent?:'flex-start'|'flex-end'|'center'|'stretch'|'space-between'|'space-around';
    direction?:'row'|'row-reverse'|'column'|'column-reverse';
    height?:string;
    margin?:TrblType;
    wrap?:boolean;
    grow?:number;
    shrink?:number;
    basis?:string;
    order?:number;
    alignSelf?:'auto'|'flex-start'|'flex-end'|'center'|'baseline'|'stretch';
};

const Box:StatelessComponent<PropsType> = (props):JSX.Element => {
    return <div className={props.className}>{props.children}</div>;
};

export { PropsType };
export default Box;
