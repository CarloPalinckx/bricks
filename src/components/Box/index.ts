export { default as default } from './Box.style';
