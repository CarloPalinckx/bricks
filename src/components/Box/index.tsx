import React, { StatelessComponent } from 'react';
import { TrblType } from '../../utility/trbl';
import StyledBox from './style';

type PropsType = JSX.IntrinsicElements['div'] & {
    justifyContent?:
        | 'flex-start'
        | 'flex-end'
        | 'center'
        | 'stretch'
        | 'space-between'
        | 'space-around'
        | 'space-evenly';
    alignItems?: 'flex-start' | 'flex-end' | 'center' | 'stretch';
    alignContent?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'space-between' | 'space-around';
    direction?: 'row' | 'row-reverse' | 'column' | 'column-reverse';
    inline?: boolean;
    height?: string;
    margin?: TrblType;
    wrap?: boolean;
    grow?: number;
    shrink?: number;
    basis?: string;
    order?: number;
    alignSelf?: 'auto' | 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch';
};

const Box: StatelessComponent<PropsType> = (props): JSX.Element => {
    const { order, direction, wrap, height, ref, ...filteredProps } = props;

    const newProps = {
        ...filteredProps,
        flexWrap: wrap,
        elementHeight: height,
        flexDirection: direction,
        flexOrder: order,
    };

    return <StyledBox {...newProps}>{props.children}</StyledBox>;
};

export { PropsType };
export default Box;
