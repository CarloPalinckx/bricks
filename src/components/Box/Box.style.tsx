import { StyledComponentClass as _S } from 'styled-components';
import _T from '../../types/ThemeType';
import styled from '../../utility/styled';
import Box from './Box.template';

const StyledBox = styled(Box)`
    display: flex;
    height: ${({ height }):string => height !== undefined ? height : 'auto' };
    flex-wrap: ${({ wrap }):string => wrap ? 'wrap' : 'no-wrap'};
    flex-direction: ${({ direction }):string => direction !== undefined ? direction : 'row' };
    justify-content: ${({ justifyContent }):string => justifyContent !== undefined ? justifyContent : 'flex-start' };
    align-items: ${({ alignItems }):string => alignItems !== undefined ? alignItems : 'stretch' };
    align-content: ${({ alignContent }):string => alignContent !== undefined ? alignContent : 'stretch' };
    margin: ${({ margin }):string => margin !== undefined ? `${margin.top} ${margin.right} ${margin.bottom} ${margin.left}` : '0' };
    flex-grow: ${({ grow }):number => grow ? grow : 0 };
    flex-shrink: ${({ shrink }):number => shrink ? shrink : 1 };
    flex-basis: ${({ basis }):string => basis ? basis : 'auto' };
    order: ${({ order }):number => order ? order : 0 };
    align-self: ${({ alignSelf }):string => alignSelf ? alignSelf : 'auto' };
`;

export default StyledBox;
