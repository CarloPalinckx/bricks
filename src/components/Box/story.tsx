import { boolean, number, select, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import styled from 'styled-components';
import Box, { PropsType } from '.';
import trbl, { PxCoordinateType } from '../../utility/trbl';
import Text from '../Text';

const Wrapper = styled.div`
    border: 1px dashed #aaa;
    height: 90vh;
`;

const Item = styled.div`
    background: #eee;
    border: 1px dashed #aaa;
    width: 100%;
    height: 100%;
`;

const justifyOptions = ['flex-start', 'flex-end', 'center', 'stretch', 'space-between', 'space-around', 'space-evenly'];
const alignItemsOptions = ['flex-start', 'flex-end', 'center', 'stretch'];
const alignContentOptions = ['flex-start', 'flex-end', 'center', 'stretch', 'space-between', 'space-around'];
const alignSelf = ['auto', 'flex-start', 'flex-end', 'center', 'baseline', 'stretch'];
const directionOptions = ['row', 'row-reverse', 'column', 'column-reverse'];

storiesOf('Box', module).add('Default', () => {
    const items: Array<JSX.Element> = [];

    for (let i = 0; i < 6; i++) {
        items.push(
            <Box
                key={i}
                margin={trbl(select('margin', [0, 6, 12], 0, 'Child') as PxCoordinateType)}
                padding={trbl(select('padding', [0, 6, 12], 0, 'Child') as PxCoordinateType)}
                grow={number('grow', 1, {}, 'Child')}
                shrink={number('shrink', 1, {}, 'Child')}
                basis={text('basis', 'auto', 'Child')}
                order={number('order', 1, {}, 'Child')}
                alignSelf={select('alignSelf', alignSelf, alignSelf[0], 'Child') as PropsType['alignSelf']}
                height={text('height', '', 'Child')}
                maxHeight={text('maxHeight', '', 'Child')}
                minHeight={text('minHeight', '', 'Child')}
                width={text('width', '', 'Child')}
                maxWidth={text('maxWidth', '', 'Child')}
                minWidth={text('minWidth', '', 'Child')}
            >
                <Item>
                    <Box padding={trbl(48)}>
                        <Text>Box {i + 1}</Text>
                    </Box>
                </Item>
            </Box>,
        );
    }

    return (
        <Wrapper>
            <Box
                margin={trbl(select('margin', [-12, -6, 0, 6, 12], 0, 'Parent') as PxCoordinateType)}
                padding={trbl(select('padding', [-12, -6, 0, 6, 12], 0, 'Parent') as PxCoordinateType)}
                justifyContent={
                    select('justifyContent', justifyOptions, justifyOptions[0], 'Parent') as PropsType['justifyContent']
                }
                alignItems={
                    select('alignItems', alignItemsOptions, alignItemsOptions[0], 'Parent') as PropsType['alignItems']
                }
                alignContent={
                    select(
                        'alignContent',
                        alignContentOptions,
                        alignContentOptions[0],
                        'Parent',
                    ) as PropsType['alignContent']
                }
                direction={
                    select('direction', directionOptions, directionOptions[0], 'Parent') as PropsType['direction']
                }
                wrap={boolean('wrap', true, 'Parent')}
                height="100%"
                width="100%"
            >
                {items}
            </Box>
        </Wrapper>
    );
});
