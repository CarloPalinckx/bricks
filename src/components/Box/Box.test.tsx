import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Box from '.';
import trbl from '../../utility/trbl';

describe('Box', () => {
    it('aligns children to the right', () => {
        const component = shallow(<Box justifyContent="flex-end" />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('aligns children in the center', () => {
        const component = shallow(<Box justifyContent="center" />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('aligns children to the bottom', () => {
        const component = shallow(<Box alignItems="flex-end" />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('aligns children in the vertical center', () => {
        const component = shallow(<Box alignItems="center" />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('can change direction', () => {
        const component = shallow(<Box direction="column" />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('can have a margin', () => {
        const component = shallow(<Box margin={trbl(24, 0, 'auto')} />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('prevents wrapping of children', () => {
        const component = shallow(<Box wrap={false} />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('stretches the children', () => {
        const component = shallow(<Box justifyContent="stretch" alignItems="stretch" alignContent="stretch" />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('can shrink', () => {
        const component = shallow(<Box shrink={1} />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('can grow', () => {
        const component = shallow(<Box grow={1} />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('can have a specific size', () => {
        const component = shallow(<Box basis="calc(25% - 24px)" />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('can be rearranged by order', () => {
        const component = shallow(<Box order={1} />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('can align itself differently then its siblings', () => {
        const component = shallow(<Box alignSelf="flex-end" />);

        expect(toJson(component)).toMatchSnapshot();
    });
});
