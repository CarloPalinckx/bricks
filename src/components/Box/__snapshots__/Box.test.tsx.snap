// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box aligns children in the center 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  margin: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<Box
  className="c0"
  justifyContent="center"
/>
`;

exports[`Box aligns children in the vertical center 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  margin: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<Box
  alignItems="center"
  className="c0"
/>
`;

exports[`Box aligns children to the bottom 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  margin: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<Box
  alignItems="flex-end"
  className="c0"
/>
`;

exports[`Box aligns children to the right 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  margin: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<Box
  className="c0"
  justifyContent="flex-end"
/>
`;

exports[`Box can align itself differently then its siblings 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  margin: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

<Box
  alignSelf="flex-end"
  className="c0"
/>
`;

exports[`Box can be rearranged by order 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  margin: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<Box
  className="c0"
  order={1}
/>
`;

exports[`Box can change direction 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  margin: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<Box
  className="c0"
  direction="column"
/>
`;

exports[`Box can grow 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  margin: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<Box
  className="c0"
  grow={1}
/>
`;

exports[`Box can have a margin 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  margin: 24px 0px auto 0px;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<Box
  className="c0"
  margin={
    Object {
      "bottom": "auto",
      "left": "0px",
      "right": "0px",
      "top": "24px",
    }
  }
/>
`;

exports[`Box can have a specific size 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  margin: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: calc(25% - 24px);
  -ms-flex-preferred-size: calc(25% - 24px);
  flex-basis: calc(25% - 24px);
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<Box
  basis="calc(25% - 24px)"
  className="c0"
/>
`;

exports[`Box can shrink 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  margin: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<Box
  className="c0"
  shrink={1}
/>
`;

exports[`Box prevents wrapping of children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  margin: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<Box
  className="c0"
  wrap={false}
/>
`;

exports[`Box stretches the children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  margin: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<Box
  alignContent="stretch"
  alignItems="stretch"
  className="c0"
  justifyContent="stretch"
/>
`;
